postgres-json-schema--0.1.1.sql

```diff
@@ -237,6 +237,28 @@ BEGIN
     END IF;
   END IF;
 
+  IF schema ? 'format' AND jsonb_typeof(data) = 'string' THEN
+    DECLARE
+      target text := (data #>> '{}');
+      EMAIL_RX constant text := '^[\w.!#$%&''*+/=?^`{|}~-]+@[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?(?:\.[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?)*$';
+    BEGIN
+      CASE (schema->>'format')
+        WHEN 'date-time' THEN PERFORM target::timestamptz; 
+        WHEN 'date'      THEN PERFORM target::date;
+        WHEN 'time'      THEN PERFORM target::time;
+        WHEN 'duration'  THEN PERFORM target::interval; 
+        WHEN 'uuid'      THEN PERFORM target::uuid;
+        WHEN 'ipv6'      THEN PERFORM target::inet; IF target NOT LIKE '%:%' THEN RAISE; END IF;
+        WHEN 'ipv4'      THEN PERFORM target::inet; IF target LIKE '%:%' THEN RAISE; END IF;
+        WHEN 'regex'     THEN PERFORM '' ~ target; 
+        WHEN 'email'     THEN IF target !~* EMAIL_RX THEN RAISE; END IF;
```

Comment:
See regex comment above, a case sensitive check would be faster:
```suggestion
        WHEN 'email'     THEN IF target !~ EMAIL_RX THEN RAISE; END IF;
```